src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs

```diff
@@ -409,48 +386,21 @@ public static object Evaluate()
                 syntaxTree = findVarNMethodCall.ReplaceVars(syntaxTree, null, null, null, elementAccessValues);
             }
 
-            expressionTree = GetExpressionFromSyntaxTree(syntaxTree);
+            expressionTree = syntaxTree.GetCompilationUnitRoot(token);
             if (expressionTree == null)
                 throw new Exception($"BUG: Unable to evaluate {expression}, could not get expression from the syntax tree");
 
-            MetadataReference[] references = new MetadataReference[]
-            {
-                MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
-                MetadataReference.CreateFromFile(typeof(Enumerable).Assembly.Location)
-            };
-
-            CSharpCompilation compilation = CSharpCompilation.Create(
-                "compileAndRunTheExpression",
-                syntaxTrees: new[] { syntaxTree },
-                references: references,
-                options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+            try {
+                var newScript = script.ContinueWith(
```

Comment:
Is re-using script like this faster, or more memory efficient? And `script` is not affected at all by `ContinueWith`?

Reply:
Memory efficient.

Reply:
I don't think it's affected.
I did a test declaring the same variable name in two ContinueWith also changing the datatype and it works correctly.

```
var newScript = Class1.script.ContinueWith(
                    "System.String Val1 = \"one\";\nSystem.Double Val2 = 2;\nreturn $\"Some {Val1} Value {Val2} End\";");
var state = await newScript.RunAsync();
Console.WriteLine("result " + state.ReturnValue);

newScript = Class1.script.ContinueWith(
                    "System.Double Val1 = 3;\nSystem.Double Val2 = 3;\nreturn $\"Some {Val1} Value {Val2} End\";");
state = await newScript.RunAsync();
Console.WriteLine("result " + state.ReturnValue);
```


Result:
result Some one Value 2 End
result Some 3 Value 3 End